backend-lib/src/main/rust/lib.rs

```diff
@@ -348,24 +348,12 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_getAccoun
             .filter(|account| matches!(account.source(), AccountSource::Derived { .. }))
             .collect::<Vec<_>>();
 
-        let first_account = accounts.first().cloned();
-
-        Ok(utils::rust_vec_to_java(
-            env,
-            accounts,
-            JNI_ACCOUNT,
-            |env, account| encode_account(env, &network, account),
-            |env| {
-                // The array contains non-null values in Kotlin. This returns `null` if
-                // there are no accounts, which is fine because the array has no entries
-                // and thus the empty element is never used.
-                match first_account {
-                    Some(account) => encode_account(env, &network, account),
-                    None => Ok(JObject::null()),
-                }
-            },
-        )?
-        .into_raw())
+        Ok(
+            utils::rust_vec_to_java(env, accounts, JNI_ACCOUNT, |env, account| {
+                encode_account(env, &network, account)
+            })?
+            .into_raw(),
+        )
```

Comment:
The only change for each call to `rust_vec_to_java` is to remove the last argument. `rustfmt` makes the changes look busier than they are. (Here we also remove `first_account`, because it is now unused.)